canopen/pdo/base.py

```diff
@@ -371,8 +371,10 @@ def _raw_from(param):
 
     def save(self) -> None:
         """Save PDO configuration for this map using SDO."""
-        logger.info("Setting COB-ID 0x%X and temporarily disabling PDO",
-                    self.cob_id)
+        if self.cob_id is None:
+            logger.info(f"Skipping saving of uninitialized PDO")
+            return
+        logger.info(f"Setting COB-ID 0x{self.cob_id} and temporarily disabling PDO")
```

Comment:
Huh?

> This PR changes all occurrences of % formatting and str.format() to f-strings.

I used f-string?

Reply:
Well, that description is not entirely accurate.  For the `logging` library, passing the substitution parameters individually is apparently recommended, because they can be processed by logging handlers.  If you only passed a ready-made string with the substitutions already made, that is no longer possible.

Thus, we should use f-strings consistently EXCEPT for calls to the logging functions.